integration/v3_election_test.go

```diff
@@ -197,3 +197,182 @@ func TestElectionSessionRecampaign(t *testing.T) {
 		t.Fatalf("expected value=%q, got response %v", "def", resp)
 	}
 }
+
+// TestElectionOnPrefixOfExistingKey checks that a single
+// candidate can be elected on a new key that is a prefix
+// of an existing key. To wit, check for regression
+// of bug #6278. https://github.com/coreos/etcd/issues/6278
+//
+// Also a good basic test of single candidate election
+// and simultaneous correct observation of an election.
+//
+func TestElectionOnPrefixOfExistingKey(t *testing.T) {
```

Comment:
This test is more complicated than it needs to be. I think this should trigger the old, broken behavior with a lot less code:
```sh
clus := NewClusterV3(t, &ClusterConfig{Size : 1})
defer clus.Terminate(t)

cli := clus.RandClient()
if _, err := cli.Put(context.TODO(), "testa"); err != nil {
    t.Fatal(err)
}
s, serr :=  concurrency.NewSession(cli)
if serr != nil {
    return serr
}
e := concurrency.NewElection(s, "test")
ctx, cancel := context.WithTimeout(context.TODO(), 5*time.Second)
err := e.Campaign(ctx, "abc")
cancel()
if err != nil {
    t.Fatal(err)
}
```